ambry-router/src/main/java/com.github.ambry.router/DeleteOperation.java

```diff
@@ -209,8 +208,9 @@ private void cleanupExpiredInflightRequests() {
       DeleteRequestInfo deleteRequestInfo = itr.next().getValue();
       if (time.milliseconds() - deleteRequestInfo.startTimeMs > routerConfig.routerRequestTimeoutMs) {
         itr.remove();
-        responseHandler
-            .onRequestResponseException(deleteRequestInfo.replica, new IOException("Timed out waiting for a response"));
+        // Do not notify this as a failure to the response handler, as this timeout could simply be due to
+        // connection unavailability. If there is indeed a network error, it will get reported eventually and the
+        // response handler will be notified accordingly.
```

Comment:
I am just feeling it is a bit vague. If instead saying `it will get reported eventually in xxx`, I would choose to add this information.